quipucords/api/scanjob/model.py

```diff
@@ -280,6 +267,28 @@ def queue(self):  # noqa: C901
         self.save()
         self.log_current_status()
 
+    def delete_inspect_results(self):
+        """Delete InspectResults exclusively related to this ScanJob."""
+        # results_from_other_jobs = InspectResult.objects.exclude(
+        #     tasks__in=self.tasks.all()
+        # ).values(
+        #     "id",
+        # )
+        # InspectResult.objects.exclude(id__in=results_from_other_jobs).filter(
+        #     tasks__job_id=self.id
+        # ).delete()
+        # TODO: why the approach above didn't work? I had to use the m2m model to do it
+        results = InspectResult.objects.filter(tasks__in=self.tasks.all())
+        ThroughModel = InspectResult.tasks.through
+        results_from_other_jobs = (
+            ThroughModel.objects.filter(inspectresult__in=results)
+            .exclude(scantask__in=self.tasks.all())
+            .values(
+                "inspectresult_id",
+            )
+        )
+        results.exclude(id__in=results_from_other_jobs).delete()
```

Comment:
```suggestion
        InspectResult.objects.filter(tasks__job_id=self.id).exclude(
            tasks__in=ScanTask.objects.exclude(job_id=self.id)
        ).delete()
```
I think this will do what you want. 🙂 I arrived at this odd-looking chain of excludes by breaking down the problem down to what I think are the basic requirements.

We want to find all `InspectResult`s that have a task that have "this" `ScanJob`. That part's easy:
```python
InspectResult.objects.filter(tasks__job_id=self.id)
```

Now, let's find all `ScanTask`s that belong do a _different_ `ScanJob`:
```python
ScanTask.objects.exclude(job_id=self.id)
```

By extending the first queryset to exclude any `InspectResult`s that have `ScanTask`s in the previous set, that effectively means we are saying "give me all results with tasks related to this job, and exclude any results that have tasks related to any _other_ jobs":
```python
InspectResult.objects.filter(tasks__job_id=self.id).exclude(tasks__in=ScanTask.objects.exclude(job_id=self.id))
```

I checked out your branch, make this change, and ran the tests in `test_scanjob.py`. Everything passes, but you may want to triple-check my work or add a couple _more_ tests out of an abundance of caution.

This new command still generates some gnarly subqueries, but I think this is near the best we can get using native queryset operations since Django refuses to implement the standard `<>` not-equals operator which I believe could simplify the generated query. 😡 Here is the raw query I produced by my queryset where the `ScanJob`'s ID is `420`:

```sql
SELECT "api_inspectresult"."id",
       "api_inspectresult"."name",
       "api_inspectresult"."status",
       "api_inspectresult"."source_id"
FROM "api_inspectresult"
INNER JOIN "api_inspectresult_tasks" ON ("api_inspectresult"."id" = "api_inspectresult_tasks"."inspectresult_id")
INNER JOIN "api_scantask" ON ("api_inspectresult_tasks"."scantask_id" = "api_scantask"."id")
WHERE ("api_scantask"."job_id" = 420
       AND NOT (EXISTS
                  (SELECT 1 AS "a"
                   FROM "api_inspectresult_tasks" V1
                   WHERE (V1."scantask_id" IN
                            (SELECT U0."id"
                             FROM "api_scantask" U0
                             WHERE NOT (U0."job_id" = 420))
                          AND V1."inspectresult_id" = ("api_inspectresult"."id"))
                   LIMIT 1)))
```

I don't _love_ it, but it seems _fine_ for now. 🤷‍♂️  By the time our tables grow large enough to cause problems with _this_ query, I'm sure we'll already have plenty of other more urgent problems _elsewhere_ in the system.